vignettes/flow-driven-spawning-decay.Rmd

```diff
@@ -0,0 +1,525 @@
+---
+title: "Spawning Decay Analysis"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Spawning Decay Analysis}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>", 
+  message = FALSE,
+  warning = FALSE
+)
+
+library(tidyverse)
+library(gt)
+library(gridExtra)
+library(lubridate)
+
+theme_set(theme_bw())
+
+```
+
+```{r setup}
+library(DSMhabitat)
+```
+
+# Objective 
+
+The objective of this analysis is to create flow-decay curves that will be applied to spawning habitat values at runtime in DSM models. The SRH2D model developed as part of the Sediment Study for the Upper Sacramento River serves as the starting point for these curves. Additional "scale-downs" are performed to reduce the magnitude of the curves in order to better match the observed relationship at each watershed where the analysis is applied. Each step in the analysis is detailed below.
+
+# SRH2D Flow-Sediment Transport Curves 
+
+Flow-to-sediment transport curves were obtained from the SRH2D model output. In total, there are 45 files, each representing a river mile. Each file contains 3 flow-to-sediment curves (in cubic meters per second) for three curve types: Parker, Wilcock, and Gaeuman. A sample output at river mile 283.330 is shown below.
+
+```{r}
+
+files_to_read <- list.files(system.file("extdata", "spawning-decay", "SedimentRatingCurves", 
+                                        package = "DSMhabitat", mustWork = TRUE),
+                            pattern = ".txt", 
+                            full.names = TRUE)
+
+rating_curves_by_rm <- map_df(files_to_read, function(x) {
+  river_mile <- str_match(basename(x), "[0-9]+\\.?[0-9]+")[,1]
+  read_tsv(x, skip = 1, col_names = c("flow", "parker_qs", "wilcock_qs", "gaeuman_qs")) |> 
+    mutate(river_mile = as.numeric(river_mile))
+})
+
+rating_curves_by_rm |> 
+  filter(river_mile == 283.330) |> 
+  knitr::kable()
+```
+
+
+All data was converted to cubic feet per second and two levels of summaries were performed. First, at each river mile, the minimum, average, and maximum values for transport were calculated across the three different curve types. Then, an average of transport was calculated for each combination of flow and minimum, mean, and maximum transport values across all the river miles. The code for the calculation is outlined below.
```

Comment:
I think this would be easier for me to interpret in a bulleted list - but fine as is if you prefer the paragraph style

Reply:
yeah this is easier to see what happens by looking at the code, I will play around with a bulleted list I think that will work!